src/tess.py

```diff
@@ -0,0 +1,364 @@
+        data_sort.reset_index(drop=True)
+
+        dt = np.diff(mjd_conversion(data_sort[time_col]))
+
+        gaps = np.where(dt > gap)[0]
+        gaps = np.append(gaps, len(data))
+        prev_idx = 0
+
+        dfs = []
+        for idx in gaps:
+            df_gap = data_sort[prev_idx : idx + 1]
+            for filt in filters:
+                df = df_gap[df_gap[filt_col] == filt].reset_index(
+                    drop=True, inplace=False
+                )
+
+                # drop the flagged or NaN values
+                if drop_bad:
+                    bad_idx = (df[flag_col] != 0) | (df[mag_col].isna())
+                    df = df[~bad_idx]
+
+                if len(df) > 0:
+                    dfs.append(df)
+
+            prev_idx = idx + 1
+
+        new_datasets = []
+        keep_columns = [
+            time_col,
+            exp_col,
+            filt_col,
+            mag_col,
+            magerr_col,
+            ra_col,
+            dec_col,
+            flag_col,
+        ]
+        for df in dfs:
+            df = df[keep_columns]
+            if len(df) > 0:
+                new_datasets.append(Lightcurve(data=df, **init_kwargs))
+
+        return new_datasets
+
+    def fetch_data_from_observatory(self, cat_row, verbose=0, **kwargs):
+        """
+        Fetch data from TESS for a given source.
+        Must return a dataframe (or equivalent), even if it is an empty one.
+        This must be implemented by each observatory subclass.
+
+        Parameters
+        ----------
+        cat_row: dict like
+            A row in the catalog for a specific source.
+            In general, this row should contain the following keys:
+            name, ra, dec, mag, filter_name (saying which band the mag is in).
+        kwargs: dict
+            Additional keyword arguments to pass to the fetcher.
+
+        Returns
+        -------
+        data : pandas.DataFrame or other data structure
+            Raw data from the observatory, to be put into a RawData object.
+        altdata: dict
+            Additional data to be stored in the RawData object.
+
+        """
+        # TODO: figure out dstArcSec threshold for flagging close stars
+        DST_THRESHOLD = 10
+        # TODO: figure out mag diff threshold for flagging similar stars
+        MAGDIFF_THRESHOLD = 0.5
+        # TODO: figure out optimal radii for queries
+        
+        # name should be source_id for GAIA catalog entry
+        name = cat_row['name']
+        dec = cat_row['dec']
+        ra = cat_row['ra']
+        mag = cat_row['mag']
+        if mag > 16:
+            # TESS can't see stars fainter than 15 mag
+            print(f"Magnitude of {mag} is too faint for TESS.")
+            return pd.DataFrame()
+
+        cone_radius = kwargs['cone_radius'] \
+                        if 'cone_radius' in kwargs else 0.01
+        sign = ' -' if dec < 0 else ' '
+        coord_str = str(ra) + sign + str(dec)
+        
+        kwargs = {
```

Comment:
This has the potential to be very confusing if `kwargs` gets used more extensively. 
Let's use `parameters` or something like that.